packages/cliqz-home-react/src/MessageCenter.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import TopMessages from './TopMessages';
+import MiddleMessages from './MiddleMessages';
+
+
+class MessageCenter extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      top: [],
+      middle: []
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let messages = nextProps.messages;
+    const topMessages = [];
+    const middleMessages = [];
+    for(var i in messages) {
+      if(messages[i].position === 'top') {
+        topMessages.push(messages[i]);
+      } else {
+        middleMessages.push(messages[i]);
+      }
+    }
+    this.setState({
+      top: topMessages,
+      middle: middleMessages
+    });
+  }
+
+  addMessages(msgs) {
```

Comment:
`addMessages(msgs = {})` so you don't have to check if msgs are there. on the other hand it is client responsibility to check what they pass